app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/components/return_banner.scala.html

```diff
@@ -16,21 +16,33 @@
 
 @import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.finance.AmountInPence
 @import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.finance.MoneyUtils
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.UserType
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.onboarding.SubscribedDetails
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.ReturnSummary
 
 @this()
 
-@(heading: String, taxOwed: AmountInPence, reference: String)(implicit messages: Messages)
+@(heading: String, returnSummary: ReturnSummary, subscribedDetails: SubscribedDetails, userType: Option[UserType])(implicit messages: Messages)
+
+@taxOwed = @{returnSummary.mainReturnChargeAmount.withFloorZero}
+@name = @{subscribedDetails.name.fold(_.value, e => e.firstName + " " + e.lastName)}
+@reference = @{if(taxOwed.isPositive) subscribedDetails.cgtReference.value else returnSummary.submissionId}
 
 <div class="govuk-box-highlight govuk-box-highlight--status">
-  <h1 class="heading-xlarge">@heading</h1>
+    @{userType match {
+        case Some(UserType.Organisation) => "Trust: " + name
```

Comment:
"Trust" and "Client" should really be read from the messages as well like:
```
message.trust.key=Trust: {0}
message.agent.key=Client: {0}
```
basically there should be no hard coded strings in the templates that get displayed to the user - when we do a translation of the service, it's the message file which will get translated so everything needs to be in there

Reply:
this still need a similar treatment that you've applied to the other template:
```scala
 @{request.userType match {
         case Some(UserType.Organisation) => messages(s"$key.organisationNameLine", name)
         case Some(UserType.Agent) => messages(s"$key.agentNameLine", name)
         case _ => name
     }}
```

Reply:
that's if we need a name for an individual - can double check if we need it here for that case?